services/blockstorage/block_sync_server.go

```diff
@@ -0,0 +1,86 @@
+package blockstorage
+
+import (
+	"errors"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
+	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+)
+
+func (s *service) sourceHandleBlockAvailabilityRequest(message *gossipmessages.BlockAvailabilityRequestMessage) error {
+	s.logger.Info("received block availability request", log.Stringable("sender", message.Sender))
+
+	lastCommittedBlockHeight := s.LastCommittedBlockHeight()
+
+	if lastCommittedBlockHeight <= message.SignedBatchRange.LastCommittedBlockHeight() {
+		return nil
+	}
+
+	firstAvailableBlockHeight := primitives.BlockHeight(1)
+	blockType := message.SignedBatchRange.BlockType()
```

Comment:
I missed this before - does this mean we're asking for results blocks and transaction blocks separately?